src/Avalonia.Diagnostics/Diagnostics/ViewModels/AvaloniaPropertyViewModel.cs

```diff
@@ -33,6 +40,21 @@ public AvaloniaPropertyViewModel(AvaloniaObject o, AvaloniaProperty property)
         public override string Name { get; }
         public override bool? IsAttached => Property.IsAttached;
         public override string Priority => _priority;
+        public override string IsPinned
+        {
+            get => _isPinned.ToString();
+            set
+            {
+                try
+                {
+                    if (Enum.TryParse(value, out PinnedStatus pinState))
+                    {
+                        _isPinned = pinState;
+                    }
+                }
+                catch { }
+            }
```

Comment:
It is bad convention to be changing the data type in the property accessors. Whatever the underlying type is really should stay exposed as the property type.